mlflow/transformers/llm_inference_utils.py

```diff
@@ -106,27 +106,34 @@ def preprocess_llm_inference_params(
             "`data` is expected to be a pandas DataFrame for MLflow inference task after signature "
             f"enforcement, but got type: {type(data)}."
         )
-
-    updated_data = []
-    params = {}
-
-    for column in data.columns:
-        if column in ["prompt", "messages"]:
-            updated_data = data[column].tolist()
-        else:
-            param = data[column].tolist()[0]
-            if column == "max_tokens":
-                params["max_new_tokens"] = param
-            elif column == "stop":
-                source_model_name = (
-                    flavor_config.get("source_model_name") if flavor_config else None
-                )
-                if stop := _get_stopping_criteria(param, source_model_name):
-                    params["stopping_criteria"] = stop
-            else:
-                params[column] = param
-
-    return updated_data, params
+    flavor_config = flavor_config or {}
+    params = params or {}
+    # Pandas convert None to np.nan internally, which is not preferred
+    data = data.replace(np.nan, None).to_dict(orient="list")
+
+    # Extract list of input data (prompt, messages) to LLM
+    task = flavor_config[_LLM_INFERENCE_TASK_KEY]
+    input_col = _LLM_INFERENCE_TASK_TO_DATA_FIELD.get(task)
+    if input_col not in data:
+        raise MlflowException(
+            f"Transformer model saved with `{task}` task excepts `{input_col}`"
+            "to be passed as input data.",
+            error_code=BAD_REQUEST,
+        )
+    update_data = data.pop(input_col)
+
+    # The rest of fields in input payload should goes to params and override default ones
+    params_in_data = {k: v[0] for k, v in data.items() if v[0] is not None}
```

Comment:
note: We only fetch params from the first input data for the batch. Not sure if it's ok, but just keeping the current behavior.

Reply:
This is OK